Parser.h

```diff
@@ -41,8 +44,20 @@ class parser
 
 
   private:
+	inline bool tryToMatchAgainstKeywords(const std::string& toReturn,
+		 const std::string& strippedLexeme,
+		 bool isLexemeQuoted,
+		 std::istream& theStream);
+	inline bool tryToMatchAgainstRegexes(const std::string& toReturn,
+		 const std::string& strippedLexeme,
+		 bool isLexemeQuoted,
+		 std::istream& theStream);
+
 	std::map<std::string, parsingFunction> registeredKeywordStrings;
+	std::map<std::string, parsingFunctionStreamOnly> registeredKeywordStringsStreamOnly;
```

Comment:
The StreamOnly register functions are before the unadorned ones, so make registeredKeywordStringsStreamOnly before registeredKeywordStrings.